From a8454b5239bbc41e36f8fcc191862355d81f0680 Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Wed, 7 Apr 2021 11:09:41 +0000
Subject: [PATCH 2/6] Fix missing function dhcp_option_name

---
 src/nm2/src/nm2_inet_state.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/src/nm2/src/nm2_inet_state.c b/src/nm2/src/nm2_inet_state.c
index 7d0f991..9afd0b0 100644
--- a/src/nm2/src/nm2_inet_state.c
+++ b/src/nm2/src/nm2_inet_state.c
@@ -109,6 +109,11 @@ bool nm2_inet_state_update(struct nm2_iface *piface)
     return true;
 }
 
+const char* dhcp_option_name(int opt_id)
+{
+    return "";
+}
+
 /**
  * Convert the interface status in @piface to a schema inet state structure
  */
-- 
1.9.1

