From c84b2cd2eababbc41316d047977ae64cd53ee000 Mon Sep 17 00:00:00 2001
From: Mateusz Bajorski <mbajorski@plume.com>
Date: Mon, 2 Mar 2020 14:24:20 +0100
Subject: [PATCH 6/6] Fix conflict with yocto-kernel-tools kconfiglib

---
 build/kconfig.mk | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/build/kconfig.mk b/build/kconfig.mk
index ba275bf..8fee772 100644
--- a/build/kconfig.mk
+++ b/build/kconfig.mk
@@ -114,7 +114,7 @@ $(KCONFIG_WORK): $(KCONFIG)
 	@echo "$(call COLOR_GREEN,Generating kconfig file and headers: $(KCONFIG_WORK))"
 	$(Q)mkdir -p "$(KWORKDIR)"
 	$(Q)$(MINIMIZE_ENV) kconfig/minimize.sh -e "$(KCONFIG)" "$(KCONFIG_WORK)"
-	$(Q)KCONFIG_CONFIG="$(KCONFIG_WORK)" genconfig kconfig/Kconfig --header-path "$(KCONFIG_WORK).h"
+	$(Q)KCONFIG_CONFIG="$(KCONFIG_WORK)" python3 -mgenconfig kconfig/Kconfig --header-path "$(KCONFIG_WORK).h"
 	@touch $(KCONFIG_WORK)
 
 .PHONY: _mk_kconfig
@@ -133,7 +133,7 @@ _kconfig_update:
 
 .PHONY: _menuconfig
 _menuconfig: $(KCONFIG_WORK)
-	$(Q)KCONFIG_CONFIG="$(KCONFIG_WORK)" menuconfig kconfig/Kconfig
+	$(Q)KCONFIG_CONFIG="$(KCONFIG_WORK)" python3 -mmenuconfig kconfig/Kconfig
 
 .PHONY: menuconfig
 menuconfig: _kconfig_target_check _menuconfig _kconfig_update
-- 
1.9.1

