From 343be69d554e36be553b7e3d4064fad42227a0ab Mon Sep 17 00:00:00 2001
From: Cyprian Lech <clech@plume.com>
Date: Mon, 26 Oct 2020 12:38:14 +0000
Subject: [PATCH 6/7] Fix xing events

---
 src/bm/src/bm_events.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/bm/src/bm_events.c b/src/bm/src/bm_events.c
index 0ffc861..e8e138b 100644
--- a/src/bm/src/bm_events.c
+++ b/src/bm/src/bm_events.c
@@ -664,12 +664,10 @@ static void
 bm_events_handle_rssi_xing_bs(bm_client_t *client, bsal_event_t *event)
 {
     bm_client_stats_t *stats;
-    bm_client_times_t  *times;
     time_t now;
     int delay;
 
     now = time(NULL);
-    times = &client->times;
     delay = 0;
 
     stats = bm_client_get_stats(client, event->ifname);
-- 
2.25.1

